db.go

```diff
@@ -571,3 +572,92 @@ func (db *DB) rewriteValidRecords(walFile *wal.WAL, validRecords []*ValueLogReco
 	}
 	return db.index.PutBatch(positions, matchKeys...)
 }
+
+func (db *DB) NewIterator(options IteratorOptions) (*MergeIterator, error) {
+	if db.options.IndexType != BTree {
```

Comment:
Maybe someday we will add a new index type which supports iterator.

```
if options.IndexType == Hash {
    return "hash index does not support iterator"
}
```

Reply:
fixed